src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,106 @@
+import classNames from 'classnames';
+import { useCallback, useMemo } from 'react';
+import { deleteTodos } from '../../api/todos';
+import { Error } from '../../types/ErrorType';
+import { FilterType } from '../../types/FilterType';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setFilterType: (filter: FilterType) => void;
+  filterType: FilterType;
+  todos: Todo[];
+  completeTodos: Todo[];
+  setSelectedTodosIds: React.Dispatch<React.SetStateAction<number[]>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorNotification: React.Dispatch<React.SetStateAction<string | null>>;
+  errorNotification: string | null;
+  selectedTodosIds: number[];
+};
+
+export const Footer: React.FC<Props> = ({
+  setFilterType,
+  filterType,
+  todos,
+  completeTodos,
+  setSelectedTodosIds,
+  setTodos,
+  setErrorNotification,
+  errorNotification,
+  selectedTodosIds,
+}) => {
+  const todosRemoved = useMemo(() => (
+    todos.filter(({ completed }) => !completed)
+  ), [todos]);
+
+  const todosCompleted = useMemo(() => (
+    todos.filter(todo => todo.completed).length),
+  [todos]);
+
+  const deleteCompletedTodo = useCallback(async () => {
+    setSelectedTodosIds(completeTodos.map(({ id }) => id));
+
+    await Promise.all(completeTodos.map(({ id }) => deleteTodos(id)))
+      .then(() => setTodos((prevTodos) => prevTodos
```

Comment:
Do not mix `async await` with `then`. Use only one of them, better `async await`.